src/modules/dashboard/components/ExpenditurePage/Stages/FormStages.tsx

```diff
@@ -38,24 +57,30 @@ const FormStages = ({
     useFormikContext<ValuesType>() || {};
   const openDeleteDraftDialog = useDialog(DeleteDraftDialog);
   const openDraftConfirmDialog = useDialog(StakeExpenditureDialog);
+  const [fieldErrorsAmount, setFieldErrorsAmount] = useState<number>(0);
 
   const handleSaveDraft = useCallback(async () => {
+    setFieldErrorsAmount(0);
     const errors = await validateForm(values);
-    const hasErrors = Object.keys(errors)?.length;
+    const errorsLength = Object.keys(errors)?.length;
     setTouched(setNestedObjectValues<FormikTouched<ValuesType>>(errors, true));
-
-    return (
-      !hasErrors &&
-      colony &&
-      openDraftConfirmDialog({
-        onClick: () => {
-          handleSubmit(values as any);
-          setActiveStateId?.(Stage.Draft);
-        },
-        isVotingExtensionEnabled: true, // 'true' is temporary value
-        colony,
-      })
+    const invalidFieldsLength: number | undefined = document
+      .getElementById('expenditurePage')
+      ?.querySelectorAll('[aria-invalid="true"]').length;
+    setFieldErrorsAmount(
+      errors && invalidFieldsLength ? invalidFieldsLength : 0,
     );
+
+    return !errorsLength && colony
+      ? openDraftConfirmDialog({
+          onClick: () => {
+            handleSubmit(values as any);
+            setActiveStateId?.(Stage.Draft);
+          },
+          isVotingExtensionEnabled: true, // 'true' is temporary value
+          colony,
+        })
+      : errorsLength;
```

Comment:
i asked about it @agnieszkajarosikloj, because this is the part of the code that was only moved by me, and she informed me that on other branch it is already changed and fixed